src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs

```diff
@@ -46,6 +77,12 @@ public AuthorizeFilter(AuthorizationPolicy policy)
                 throw new ArgumentNullException(nameof(context));
             }
 
+            Policy = Policy ?? await AuthorizationPolicy.CombineAsync(PolicyProvider, AuthorizeData);
```

Comment:
@davidfowl The lifetime of the PolicyProvider is transient.  It sounds like AuthorizeFilters live for the lifetime of the app.

@javiercn The main intent of the IPolicyProvider interface was to allow arbitrary code based generation of requirements based on the policy name. See https://github.com/aspnet/Security/issues/670.  But I would think in general this mapping would be static and rarely (if ever) change, @blowdart thoughts?